### PR TITLE
Introduce additional linters & bump golangci-lint to 1.50.0

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

 

 
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
 - Introduce additional linters
 - bump golangci-lint to 1.50.0

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Introduce additional linters & bump golangci-lint to 1.50.0
```

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -7,13 +7,17 @@ run:
 linters:
   disable-all: true
   enable:
+    - asasalint
     - asciicheck
     - bodyclose
     - containedctx
+    - decorder
```
Comment (mboersma) [inline .golangci.yml:14]: You can work around this by adding "decorder" to the .codespellignore file.

**Review by mboersma — COMMENTED** [review]
Here are the descriptions of the new linters this PR proposes to add:
- asasalint: check for pass []any as any in variadic func(...any) [fast: false, auto-fix: false]
- decorder: check declaration order and count of types, constants, variables and functions [fast: true, auto-fix: false]
- errchkjson: Checks types passed to the json encoding functions. Reports unsupported types and optionally reports occasions, where the check for the returned error can be omitted. [fast: false, auto-fix: false]
- execinquery: execinquery is a linter about query string checker in Query function which reads your Go src files and warning it finds [fast: false, auto-fix: false]
- nosprintfhostport: Checks for misuse of Sprintf to construct a host with port in a URL. [fast: true, auto-fix: false]
- reassign: Checks that package variables are not reassigned [fast: false, auto-fix: false]
- usestdlibvars: A linter that detect the possibility to use variables/constants from the Go standard library. [fast: true, auto-fix: false]

Those all sound reasonable and useful to me, although in practice it looks like all they caught were opportunities for constant usage like `200` -> `http.StatusOK`.

File: .golangci.yml (hunk 1 above)
Comment (Prajyot-Parab) [reply .golangci.yml:14]: Done.

**Comment by Prajyot-Parab** [comment]
> Here are the descriptions of the new linters this PR proposes to add:
> 
> * asasalint: check for pass []any as any in variadic func(...any) [fast: false, auto-fix: false]
> * decorder: check declaration order and count of types, constants, variables and functions [fast: true, auto-fix: false]
> * errchkjson: Checks types passed to the json encoding functions. Reports unsupported types and optionally reports occasions, where the check for the returned error can be omitted. [fast: false, auto-fix: false]
> * execinquery: execinquery is a linter about query string checker in Query function which reads your Go src files and warning it finds [fast: false, auto-fix: false]
> * nosprintfhostport: Checks for misuse of Sprintf to construct a host with port in a URL. [fast: true, auto-fix: false]
> * reassign: Checks that package variables are not reassigned [fast: false, auto-fix: false]
> * usestdlibvars: A linter that detect the possibility to use variables/constants from the Go standard library. [fast: true, auto-fix: false]
> 
> Those all sound reasonable and useful to me, although in practice it looks like all they caught were opportunities for constant usage like `200` -> `http.StatusOK`.

Thanks @mboersma for adding the detailed description of newly introduced linters.

**Review by mboersma — COMMENTED** [review]
/lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

thanks!

**Pull request merged**